### PR TITLE
change: [M3-7145] - Update DC Specific Pricing Notices

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Updates the DC specific pricing notices on Linode Create, Kubernetes Create, and NodeBalancer Create.

## Major Changes 🔄

- No notice is shown by default (this is new)
- A new **warning** notice shows when you select a region that has DC specific pricing
- No changes to Linode Clone warning (you should see `The selected region has a different price structure` if you clone to a region with different pricing)

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-09-15 at 5 08 04 PM | Screenshot 2023-09-15 at 5 07 56 PM |
| Screenshot 2023-09-15 at 5 07 28 PM | Screenshot 2023-09-15 at 5 07 19 PM |
| Screenshot 2023-09-15 at 5 07 00 PM | Screenshot 2023-09-15 at 5 06 39 PM |

### Setup
- Turn on the MSW
- Turn on the DC Specific Pricing Flag

### Linode Create
- Go to `/linodes/create`
- Select `Jakarta, ID (id-cgk)`
- Verify you see a Notice saying `Prices for plans, products, and services in Jakarta, ID may vary from other regions.`
- Select `Sao Paulo, BR (br-gru)`
- Verify you see a Notice saying `Prices for plans, products, and services in Sao Paulo, BR may vary from other regions.`
- Verify that normal regions have no notice

### Linode Clone
- Go to `/linodes/create`
- Go to the Clone tab
- Select a Linode
- Select a region with different pricing
- Verify you see `The selected region has a different price structure. ` when you select a region with different pricing

### NodeBalancer Create
- Go to `/nodebalancers/create`
- Select `Jakarta, ID (id-cgk)`
- Verify you see a Notice saying `Prices for plans, products, and services in Jakarta, ID may vary from other regions.`
- Select `Sao Paulo, BR (br-gru)`
- Verify you see a Notice saying `Prices for plans, products, and services in Sao Paulo, BR may vary from other regions.`
- Verify that normal regions have no notice

### Kubernetes Create
- Go to `/kubernetes/create`
- Select `Jakarta, ID (id-cgk)`
- Verify you see a Notice saying `Prices for plans, products, and services in Jakarta, ID may vary from other regions.`
- Select `Sao Paulo, BR (br-gru)`
- Verify you see a Notice saying `Prices for plans, products, and services in Sao Paulo, BR may vary from other regions.`
- Verify that normal regions have no notice

### Other Testing
- Verify that no DC Specific pricing notices show when the feature flag is off
- Unit tests:
```
yarn test SelectRegionPanel.test.tsx linodes.test.ts 
```

### CODE REVIEW

File: packages/manager/src/utilities/pricing/linodes.test.ts [hunk 1]
```diff
@@ -232,3 +233,64 @@ describe('isLinodeTypeDifferentPriceInSelectedRegion', () => {
...
+    });
+    expect(doesRegionHaveUniquePricing('us-east', types)).toBe(false);
+  });
+  it('should return false when no linode type has unique pricing for the specificed region', () => {
+    const types = linodeTypeFactory.buildList(1, {
+      region_prices: [{ id: 'id-cgk' }],
+    });
+    expect(doesRegionHaveUniquePricing('us-east', types)).toBe(false);
+  });
+  it('should return true when any linode type has unique pricing for the specificed region', () => {
```
Comment (mjac0bs) [inline packages/manager/src/utilities/pricing/linodes.test.ts:290]:
```suggestion
  it('should return true when any linode type has unique pricing for the specified region', () => {
```

File: packages/manager/src/utilities/pricing/linodes.test.ts [hunk 2]
```diff
@@ -232,3 +233,64 @@ describe('isLinodeTypeDifferentPriceInSelectedRegion', () => {
...
+    });
+    expect(doesRegionHaveUniquePricing('us-east', types)).toBe(false);
+  });
+  it('should return false when no region has specific pricing', () => {
+    const types = linodeTypeFactory.buildList(1, {
+      region_prices: [],
+    });
+    expect(doesRegionHaveUniquePricing('us-east', types)).toBe(false);
+  });
+  it('should return false when no linode type has unique pricing for the specificed region', () => {
```
Comment (mjac0bs) [inline packages/manager/src/utilities/pricing/linodes.test.ts:284]:
```suggestion
  it('should return false when no linode type has unique pricing for the specified region', () => {
```

Comment (mjac0bs) [reply packages/manager/src/utilities/pricing/linodes.test.ts:284]: Specified or specific? 😄 Either would work, just not a combination.

File: packages/manager/src/utilities/pricing/linodes.ts [hunk 3]
```diff
@@ -129,3 +129,25 @@ export const getDynamicDCNetworkTransferData = ({
     used: networkTransferData.used || 0,
   };
 };
+
+/**
+ * This function is used to determine if specific pricing exists in a region
+ * given any number of Linode Types
+ * @param regionId the region to check for specific pricing
+ * @param types an array of Linode Types
+ * @returns true if there is at least one linode type with a DC sprcific price for the provided region
```
Comment (mjac0bs) [inline packages/manager/src/utilities/pricing/linodes.ts:138]:
```suggestion
 * @returns true if there is at least one linode type with a DC specific price for the provided region
```

**Review by mjac0bs — COMMENTED** [review]
Nice testing instructions! Tested the various flows and found one place (NodeBalancer create) where we're missing a feature flag check. Confirmed that no notices appear on Databases, as expected, since it is out of scope for DC pricing.

Otherwise, this is looking good pending that feedback is addressed and typos are corrected.

Yeah, this makes sense, showing the region pricing notice if the linode we're cloning from has region specific pricing (e.g. is `id-cgk`). If we change to any other region with different prices, then the clone notice for different prices displays instead. 👍🏼 
Screenshot 2023-09-18 at 10 42 42 AM

File: packages/manager/src/components/SelectRegionPanel/SelectRegionPanel.tsx [hunk 4]
```diff
@@ -46,25 +51,48 @@ export const SelectRegionPanel = (props: SelectRegionPanelProps) => {
   const flags = useFlags();
   const params = getQueryParamsFromQueryString(location.search);
 
+  const isCloning = /clone/i.test(params.type);
```
Comment (jaalah-akamai) [inline packages/manager/src/components/SelectRegionPanel/SelectRegionPanel.tsx:54]: Nice 👍

**Review by jaalah-akamai — APPROVED** [review]
Approved. Tested that the notices only appear for the select regions using test instructions

**Pull request merged**